status-im/status-react
# Add linter for inconsistent deftest name

### Summary

A quick win :) This PR adds a new linter to verify all test names are consistent with one common convention we already follow, for the most part, which is:

> Test vars (test names) should be suffixed with `-test`.

My motivation was that I was going to make again another comment about this in a PR review, but then I realized it wouldn't take much longer to just lint this problem once and for all.

There's no strong reason for following this convention, although it's quite common in Clojure, but in any case, these are the reasons I can think of and remember:

- Naming consistency. Sometimes tests start with `test-`, others end with `test` and others don't prefix/suffix at all. While fixing lint errors in this PR I saw a few namespaces that had all 3 conventions in the same file.
- The suffix removes potential conflicts with core Clojure functions.
- The suffix mostly eliminates potential conflicts with other vars in the test namespace. Example: you can declare a function `delete` and have a test named `delete-test`.
- For someone using Emacs `imenu` feature, it helps differentiate which vars are tests and which are just local functions supporting the tests.

#### Areas that may be impacted

None.

### Steps to test

For a dev, just open any test namespace and try to give it an "invalid" name and the linter error should be displayed. If it doesn't, you may need to remove the `.clj-kondo/cache` directory and even restart your editor (I had to). `make lint` should work as expected, this potential problem is just for the live editor preview of errors.

status: ready
